fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -52,12 +52,33 @@ def self.run(params)
         UI.message(message.join(" "))
 
         app.tunes_all_build_trains(platform: platform).each do |train|
+          # We try to check first with version_string, but also we try to check with display_version_string since
+          # this value could be different.
+          #
+          # Example:
+          # - App Store version set as 1.0 and app version set as 1.0.0;
+          #
+          # In that case, the response from API will be "versionString = 1.0" and "displayVersionString = 1.0.0"
           UI.verbose("Found train: #{train.version_string}, comparing to supplied version: #{version}")
-          if version && version != train.version_string
+          if version && train.version_string == train.display_version_string && version != train.version_string
```

Comment:
I think instead of adding the additional conditions in this `if` and the `else`, we could just change the original from `if version && version != train.version_string` to `if version && version != train.display_version_string`

Both of the changes that were done essentially come down to `version != train.display_version_string` so it doesn't look like there is a reason to have extra comparisons (except for maybe the verbose logging???)

Reply:
I did my first implementation like that, I just wanted to play safe as I don't know if keep only one condition will fix the problem and continue working for all users of Fastlane.